client/my-sites/post-type-filter/index.jsx

```diff
@@ -5,27 +5,74 @@ import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
 import map from 'lodash/map';
 import find from 'lodash/find';
+import get from 'lodash/get';
 import includes from 'lodash/includes';
 
 /**
  * Internal dependencies
  */
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { getSiteSlug } from 'state/sites/selectors';
+import { getSiteSlug, isJetpackSite } from 'state/sites/selectors';
 import { getAllPostCounts } from 'state/posts/counts/selectors';
+import { mapPostStatus } from 'lib/route/path';
 import UrlSearch from 'lib/mixins/url-search';
 import QueryPostCounts from 'components/data/query-post-counts';
 import SectionNav from 'components/section-nav';
 import NavTabs from 'components/section-nav/tabs';
 import NavItem from 'components/section-nav/item';
 import Search from 'components/search';
 
+/**
+ * Constants
+ */
+const POST_STATUSES = [ 'publish', 'draft', 'pending', 'private', 'future', 'trash' ];
+
+/**
+ * Utility
+ */
+
+/**
+ * Returns an object of normalized post counts, summing publish/private and
+ * pending/draft counts.
+ *
+ * @param  {Object} counts Original post counts
+ * @return {Object}        Normalized post counts
+ */
+function getNormalizedCounts( counts ) {
+	return POST_STATUSES.reduce( ( memo, status ) => {
+		const count = get( counts, status, 0 );
+
+		let key;
+		switch ( status ) {
+			case 'publish':
+			case 'private':
+				key = 'publish';
+				break;
+
+			case 'draft':
+			case 'pending':
+				key = 'draft';
+				break;
+
+			default:
+				key = status;
+		}
+
+		return {
+			...memo,
+			[ key ]: ( memo[ key ] || 0 ) + count
+		};
+	}, {} );
+}
+
```

Comment:
This seems a bit out of place here.  I am thinking it would be better as a selector `getNormalizedPostCounts` or similar, which would replace `getAllPostCounts` below.  It doesn't look there is any usage of the non-normalized data within `counts`, just truthiness tests.